src/ray/raylet/scheduling_resources.cc

```diff
@@ -302,6 +377,57 @@ std::string ResourceIds::ToString() const {
   return return_string;
 }
 
+void ResourceIds::UpdateCapacity(int64_t new_capacity) {
+  // Assert the new capacity is positive for sanity
+  RAY_CHECK(new_capacity >= 0);
+  int64_t capacity_delta = new_capacity - total_capacity_;
```

Comment:
I'm confused about this. Is `total_capacity_` always an int? It's currently declared as a double.

Reply:
`total_capacity_` is double because by definition, it should include fractional resources. I could change it to int to reflect it's current usage, but I think it's better to leave it as a double so it stays true to it's definition. Thanks for pointing this out though, I've updated this line to `int64_t capacity_delta = new_capacity - (int64_t)total_capacity_;` to truncate any fractional resources from this calculation.

Reply:
`total_capacity_` is initialized to an integer, right? And it's only updated in `UpdateCapacity`, which requires an integer value, right? So when could it be not an integer?

Also, the line `int64_t capacity_delta = new_capacity - (int64_t)total_capacity_;` really concerns me because if `total_capacity_` has some fractional value, then we are truncating it and losing that value. Wouldn't that lead to incorrect behavior?

Also, if we do need to cast it, please right it as `int64_t capacity_delta = new_capacity - static_cast<int64_t>(total_capacity_);` to avoid C-style casts.

Reply:
`total_capacity_` is not necessarily initialized as an integer - it may be initialized as a double when a `ResourceIds` object is [instantiated with fractional resources](https://github.com/romilbhardwaj/ray/blob/ce30cdbc1238d21787a0dba17504d09c93ebdc7a/src/ray/raylet/scheduling_resources.cc#L238-L248). 

I think the cast to int is required in `int64_t capacity_delta = new_capacity - (int64_t)total_capacity_;` for correct behavior. Consider the two possible cases:
1) `total_capacity_` is a whole number -> there are no fractional resources. The cast has no effect.
2) `total_capacity_` is not a whole number -> there exist fractional resources. If we do not do the explicit `int64_t` cast, the subtraction `new_capacity - total_capacity_` followed by a `int64_t` cast for `capacity_delta` would result in an incorrect result. For instance, consider new_capacity = 8 and total_capacity_ = 7.2. The value of `capacity_delta` after evaluating `int64_t capacity_delta = new_capacity - (int64_t)total_capacity_;` would be 0, whereas we would want it to be 1 (and thus the resultant capacity would be 8.2).